dvc/dependency/repo.py

```diff
@@ -42,17 +49,36 @@ def _make_repo(self, **overrides):
         with external_repo(**merge(self.def_repo, overrides)) as repo:
             yield repo
 
+    def _git_status(self):
+        cache_dir = self.repo.cache.local.cache_dir
+        with self._make_repo(
+            cache_dir=os.path.join(cache_dir, "old")
+        ) as old_repo:
+            with self._make_repo(
+                cache_dir=os.path.join(cache_dir, "new"), rev_lock=None
```

Comment:
Same
```suggestion
                rev_lock=None
```

Reply:
Without `cache_dir` specified explicitly, both repos seem to be pulled into the same directory, and so the update to the imported file is never detected.

Reply:
@Baranowski But cache_dir has nothing to do with the update, right? So I suppose the reason might be a bug in the caching system in dvc/external_repo.py?

Reply:
You are right, @efiop, I think I can see the bug in `external_repo.py`

Reply:
@Baranowski please see https://github.com/iterative/dvc/pull/2889#discussion_r357951525 . Let's create a separate ticket.